src/algokit_utils/applications/app_client.py

```diff
@@ -1682,18 +1615,22 @@ def get_line_for_pc(input_pc: int) -> int | None:
                     get_line_for_pc=custom_get_line_for_pc,
                     traces=None,
                 )
-
         if error_message:
             import re
 
             message = e.logic_error_str if isinstance(e, LogicError) else str(e)
             app_id = re.search(r"(?<=app=)\d+", message)
             tx_id = re.search(r"(?<=transaction )\S+(?=:)", message)
-            error = Exception(
+            runtime_error_message = (
                 f"Runtime error when executing {app_spec.name} "
                 f"(appId: {app_id.group() if app_id else 'N/A'}) in transaction "
                 f"{tx_id.group() if tx_id else 'N/A'}: {error_message}"
             )
+            if isinstance(e, LogicError):
+                e.message = runtime_error_message
+                error = e
+            else:
+                e = Exception(runtime_error_message)
             error.__cause__ = e
```

Comment:
In the case of returning a logic error, it's a little weird for the `__cause__` to be itself. I'm not sure if this was done for compatibility reasons?